package/src/components/MessageMenu/MessageUserReactions.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useMemo } from 'react';
+import { FlatList, StyleSheet, Text, View } from 'react-native';
+
+import { ReactionSortBase } from 'stream-chat';
+
+import { useFetchReactions } from './hooks/useFetchReactions';
+import { ReactionButton } from './ReactionButton';
+
+import {
+  MessagesContextValue,
+  useMessagesContext,
+} from '../../contexts/messagesContext/MessagesContext';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';
+import { DefaultStreamChatGenerics, Reaction } from '../../types/types';
+import { ReactionData } from '../../utils/utils';
+import { MessageType } from '../MessageList/hooks/useMessageList';
+
+export type MessageUserReactionsProps<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = Partial<
+  Pick<
+    MessagesContextValue<StreamChatGenerics>,
+    'MessageUserReactionsAvatar' | 'MessageUserReactionsItem' | 'supportedReactions'
+  >
+> & {
+  /**
+   * The message object
+   */
+  message?: MessageType<StreamChatGenerics>;
+  /**
+   * An array of reactions
+   */
+  reactions?: Reaction[];
+};
+
+const sort: ReactionSortBase = {
+  created_at: -1,
+};
+
+export const MessageUserReactions = (props: MessageUserReactionsProps) => {
+  const {
+    message,
+    MessageUserReactionsAvatar: propMessageUserReactionsAvatar,
+    MessageUserReactionsItem: propMessageUserReactionsItem,
+    reactions: propReactions,
+    supportedReactions: propSupportedReactions,
+  } = props;
+  const reactionTypes = Object.keys(message?.reaction_groups ?? {});
+  const [selectedReaction, setSelectedReaction] = React.useState<string | undefined>(
+    reactionTypes[0],
+  );
+  const {
+    MessageUserReactionsAvatar: contextMessageUserReactionsAvatar,
+    MessageUserReactionsItem: contextMessageUserReactionsItem,
+    supportedReactions: contextSupportedReactions,
+  } = useMessagesContext();
+  const supportedReactions = propSupportedReactions ?? contextSupportedReactions;
+  const MessageUserReactionsAvatar =
+    propMessageUserReactionsAvatar ?? contextMessageUserReactionsAvatar;
+  const MessageUserReactionsItem = propMessageUserReactionsItem ?? contextMessageUserReactionsItem;
+
+  const messageReactions = reactionTypes.reduce<ReactionData[]>((acc, reaction) => {
```

Comment:
How large is `reactionTypes.length` ? I don't think it's too big, but we are still running this `.reduce` on every render. Should probably be wrapped in `useMemo`.

Reply:
Good point. I have added it.